monster-cards/src/widgets/navbar/createNavbar.ts

```diff
@@ -1,41 +1,44 @@
-import createRenderableChildrenMixin from 'dojo-widgets/mixins/createRenderableChildrenMixin';
-import createRenderMixin from 'dojo-widgets/mixins/createRenderMixin';
-import createStatefulChildrenMixin from 'dojo-widgets/mixins/createStatefulChildrenMixin';
-
-import createNavMenu from './createNavMenu';
-import createNavActions from './createNavActions';
-
-const createNavbar = createRenderMixin
-	.mixin(createRenderableChildrenMixin)
-	.mixin({
-		mixin: createStatefulChildrenMixin,
-		initialize(instance) {
-			instance
-				.createChildren({
-					navMenu: {
-						factory: createNavMenu,
-						options: {
-							state: {
-								classes: [ 'inline-list' ]
-							}
-						}
-					},
-					navActions: {
-						factory: createNavActions,
-						options: {
-							state: {
-								classes: [ 'inline-list', 'pull-right' ]
-							}
-						}
-					}
-				})
-				.then(() => {
-					instance.invalidate();
-				});
+import { Widget, DNode } from 'dojo-interfaces/widgetBases';
+import createWidgetBase from 'dojo-widgets/bases/createWidgetBase';
+import d from 'dojo-widgets/util/d';
+import createSearchInput from '../common/createSearchInput';
+
+export type NavBarState = {
+	sections: NavBarLinkDefinition[];
+}
+
+export type NavBar = Widget<NavBarState>
+
+export type NavBarLinkDefinition = {
+	text: string;
+	href: string;
+}
+
+function listItem(childNode: DNode): DNode {
+	return d('li', {}, [ childNode ]);
+}
+
+function createNavBarLink({ text: innerHTML, href }: NavBarLinkDefinition): DNode {
+	return listItem(d('a', { href, innerHTML }));
+}
+
+const createNavbar = createWidgetBase.extend({
+	tagName: 'header.navbar',
+	childNodeRenderers: [
+		function(this: NavBar): DNode[] {
+			const homeLink = listItem(d('a', { href: '#' }, [
```

Comment:
I don't like this, I would rather is all be done in `listItem` not half `d` as param then `d` in the function also